### PR TITLE
Fix `listApplicationsThirdParty` test

### LINKED ISSUE #14250: `listApplicationsThirdParty` test is unstable
### Describe the bug

This is a tracker for intermittent failures of `AccountRestServiceTest.listApplicationsThirdParty`.

Failed runs:
* https://github.com/keycloak/keycloak/runs/8224552577?check_suite_focus=true
* https://github.com/keycloak/keycloak/runs/8213218160?check_suite_focus=true

### Version

main

### Expected behavior

_No response_

### Actual behavior

```
[ERROR] Failures:
[ERROR]   AccountRestServiceWithUserProfileTest>AccountRestServiceTest.listApplicationsThirdParty:966 expected:<[]> but was:<[web-origins]>
```

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by vmuzikar**
Closes #14250

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/account/AccountRestServiceTest.java [hunk 1]
```diff
@@ -963,7 +963,9 @@ public void listApplicationsThirdParty() throws Exception {
         assertFalse(app.getConsent().getGrantedScopes().isEmpty());
         ConsentScopeRepresentation grantedScope = app.getConsent().getGrantedScopes().get(0);
         assertEquals(clientScopeRepresentation.getId(), grantedScope.getId());
-        assertEquals(clientScopeRepresentation.getAttributes().get(ClientScopeModel.CONSENT_SCREEN_TEXT) != null ? clientScopeRepresentation.getAttributes().get(ClientScopeModel.CONSENT_SCREEN_TEXT) : clientScopeRepresentation.getName(), grantedScope.getName());
```
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/account/AccountRestServiceTest.java:966]: This is rather weird, not sure what's going on here.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/account/AccountRestServiceTest.java [hunk 2]
```diff
@@ -963,7 +963,9 @@ public void listApplicationsThirdParty() throws Exception {
...
+
+        String consentScreenText = clientScopeRepresentation.getAttributes().get(ClientScopeModel.CONSENT_SCREEN_TEXT);
+        assertEquals(consentScreenText != null && !consentScreenText.isEmpty() ? consentScreenText : clientScopeRepresentation.getName(), grantedScope.getName());
```
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/account/AccountRestServiceTest.java:968]: This is weird, as why would the consentScreenText be null sometimes, and sometimes not?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/account/AccountRestServiceTest.java (hunk 1 above)
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/account/AccountRestServiceTest.java:966]: This is weird, as it seems like it tries to do the same thing twice?

**Comment by vmuzikar** [comment]
@stianst Thanks for the review!

The test is apparently (and AFAICT by design) non-deterministic, working with a different client scope every run. The REST API is correctly not setting the consent text as a name when it's an empty string. I fixed the test to expect this behaviour.

But yeah, the test is weird and broken from my PoV due to its non-deterministic nature. Maybe @mposolda might have some insights as he reviewed the [PR](https://github.com/keycloak/keycloak/pull/13160) that broke it.

**Comment by stianst** [comment]
@vmuzikar it's probably stuff like:

https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/account/AccountRestServiceTest.java#L931

and 

https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/account/AccountRestServiceTest.java#L963

That are the issues here. There's no guarantee on the ordering there afaik, and the test should rather fetch a specific entry rather than just the first entry. That would allow the test to be changed to be deterministic, which really is what is needed here to make it stable.

**Comment by vmuzikar** [comment]
@stianst Yes, that was mine thinking too. But I figured it's non-deterministic for a reason probably as multiple tests are implemented this way. I wanted to wait for a PR review to confirm/diprove this. :)

**Comment by stianst** [comment]
> @stianst Yes, that was mine thinking too. But I figured it's non-deterministic for a reason probably as multiple tests are implemented this way. I wanted to wait for a PR review to confirm/diprove this. :)

I betja it's non-deterministic due to laziness only ;)

**Comment by vmuzikar** [comment]
I made the test deterministic by splitting it in two.

I didn't touch other tests there even though they are kinda non-deterministic too (fetching random client scopes). But in these cases it should be fine as the tests really don't care about getting a specific client scopes.

**Review by stianst — APPROVED** [review]

**Pull request merged**